hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/ContractFeature.java

```diff
@@ -0,0 +1,196 @@
+    }
+
+    @Then("the mirror node REST API should verify the deployed contract entity")
+    public void verifyDeployedContractMirror() {
+        verifyContractFromMirror(false);
+    }
+
+    @Then("the mirror node REST API should verify the deleted contract entity")
+    public void verifyDeletedContractMirror() {
+        verifyContractFromMirror(true);
+    }
+
+    private void verifyCreateContractNetworkResponse() {
+        assertNotNull(networkTransactionResponse.getTransactionId());
+        assertNotNull(networkTransactionResponse.getReceipt());
+        contractId = networkTransactionResponse.getReceipt().contractId;
+        assertNotNull(contractId);
+    }
+
+    private void persistContractBytes(String contractContents) {
+        // rely on SDK chunking feature to upload larger files
+        networkTransactionResponse = fileClient.createFile(new byte[] {});
+        assertNotNull(networkTransactionResponse.getTransactionId());
+        assertNotNull(networkTransactionResponse.getReceipt());
+        fileId = networkTransactionResponse.getReceipt().fileId;
+        assertNotNull(fileId);
+        log.info("Created file {} to hold contract init code", fileId);
+
+        networkTransactionResponse = fileClient.appendFile(fileId, contractContents.getBytes(StandardCharsets.UTF_8));
+        assertNotNull(networkTransactionResponse.getTransactionId());
+        assertNotNull(networkTransactionResponse.getReceipt());
+    }
+
+    private void createContract(String byteCode, int initialBalance) {
+        persistContractBytes(byteCode.replaceFirst("0x", ""));
+        networkTransactionResponse = contractClient.createContract(
+                fileId,
+                750000,
+                initialBalance == 0 ? null : Hbar.fromTinybars(initialBalance),
+                null);
+
+        verifyCreateContractNetworkResponse();
+    }
+
+    private MirrorContractResponse verifyContractFromMirror(boolean isDeleted) {
+        MirrorContractResponse mirrorContract = mirrorClient.getContractInfo(contractId.toString());
+
+        assertNotNull(mirrorContract);
+        assertThat(mirrorContract.getAdminKey()).isNotNull();
+        assertThat(mirrorContract.getAutoRenewPeriod()).isNotNull();
+        assertThat(mirrorContract.getBytecode()).isNotBlank();
+        assertThat(mirrorContract.getContractId()).isEqualTo(contractId.toString());
+        assertThat(mirrorContract.getCreatedTimestamp()).isNotBlank();
+        assertThat(mirrorContract.isDeleted()).isEqualTo(isDeleted);
+        assertThat(mirrorContract.getFileId()).isEqualTo(fileId.toString());
+        assertThat(mirrorContract.getMemo()).isNotBlank();
+        String address = mirrorContract.getSolidityAddress();
+        assertThat(address).isNotBlank();
+        assertThat(address).isNotEqualTo("0x");
+        assertThat(address).isNotEqualTo("0x0000000000000000000000000000000000000000");
```

Comment:
Should chain assertions to the same object:
```suggestion
        assertThat(address).isNotBlank().isNotEqualTo("0x").isNotEqualTo("0x0000000000000000000000000000000000000000");
```

Reply:
Updated